src/meshes/t8code_mesh.jl

```diff
@@ -294,13 +361,41 @@ conforming mesh from a `p4est_connectivity` data structure.
 - `initial_refinement_level::Integer`: refine the mesh uniformly to this level before the simulation starts.
 """
 function T8codeMesh{NDIMS}(conn::Ptr{p4est_connectivity}; kwargs...) where {NDIMS}
-    @assert NDIMS == 2 # Only support for NDIMS = 2 yet.
+    @assert NDIMS == 2 # Only support for NDIMS = 2.
 
     cmesh = t8_cmesh_new_from_p4est(conn, mpi_comm(), 0)
 
     return T8codeMesh{NDIMS}(cmesh; kwargs...)
 end
 
+"""
+    T8codeMesh{NDIMS}(conn::Ptr{p8est_connectivity},
+                      mapping=nothing, polydeg=1, RealT=Float64,
+                      initial_refinement_level=0)
+
+Main mesh constructor for the `T8codeMesh` that imports an unstructured,
+conforming mesh from a `p4est_connectivity` data structure.
```

Comment:
Ah, so perhaps my suggestions above in the elixirs will not work regarding the use of a different mesh file format. A new constructor would need to be written in that case yes? From what I can tell poking around the t8code repo. it can parse gmsh files saved in the `.msh` format. Would this be worthwhile to support and/or make things more user friendly?

Reply:
The user-friendly mesh constructor would be using the one for the Gmsh file input. Supporting p4est/p8est connectivities as input is merely for the testing framework so that I can recycle the P4estMesh elixiers.

Reply:
@andrewwinters5000 Maybe I should note that of course there already is a constructor for Gmsh files. :)

https://github.com/trixi-framework/Trixi.jl/blob/1421434624a8ad9bd7878a4a576676027851ca34/src/meshes/t8code_mesh.jl#L446

Reply:
I added a todo to https://github.com/trixi-framework/Trixi.jl/issues/1556 to think about a nicer mesh reader API.